mountpoint-s3/src/prefetch.rs

```diff
@@ -199,6 +216,11 @@ where
     }
 }
 
+type ParsedMetadata = Arc<RwLock<Option<HashMap<(ColumnIndex, RowGroupIndex), (u64, u64)>>>>;
```

Comment:
Ah, I see below that it's a bytes range. In that case we can use [`Range`](https://doc.rust-lang.org/std/ops/struct.Range.html) struct and it would be clearer.

Reply:
I'd recommend adding Rustdoc here so its explicit too what the `Range` is.

Reply:
Will add some comments and reference to the Rustdoc, although in the newest implementation, this has changed, and will now be an Interval Tree instead of a HashMap!